features/FEATURE_BLE/ble/Gap.h

```diff
@@ -2102,6 +2102,28 @@ class Gap {
         return rc;
     }
 
+    /**
+     * Set the parameters used during a scan procedure.
+     *
+     * @param[in] scanningParams Parameter struct containing the interval, period,
+     * timeout and active scanning toggle.
+     *
+     * @return BLE_ERROR_NONE if the scan parameters were correctly set.
+     *
+     * @note All restrictions from setScanParams(uint16_t, uint16_t, uint16_t, bool) apply.
+     */
+    ble_error_t setScanParams(const GapScanningParams& scanningParams) {
+        ble_error_t rc;
+        if (((rc = _scanningParams.setInterval(scanningParams.getInterval())) == BLE_ERROR_NONE) &&
```

Comment:
Could you delegate the call to `setScanParams(uint16_t, uint16_t, uint16_t, bool)` ?